activesupport/lib/active_support/message_encryptor.rb

```diff
@@ -1,6 +1,12 @@
 require 'openssl'
 require 'base64'
 require 'active_support/core_ext/array/extract_options'
+require 'active_support/core_ext/securerandom'
+require 'active_support/core_ext/hash/keys'
+require 'active_support/claims'
+require 'active_support/message_verifier'
+require 'active_support/encryptor/agcm'
```

Comment:
This is where you're getting circular require warnings.

`agcm` requires `jwe` upfront, which on line 7 mentions `class MessageEncryptor`. But this `message_encryptor.rb` file hasn't finished loading yet, so Rails' autoloading sees `ActiveSupport::MessageEncryptor` and tries to load the file we're already loading.